apps/worker/src/monitor/tx.status.monitor.service.ts

```diff
@@ -0,0 +1,130 @@
+    } catch (e) {
+      // 🐂 //
+    }
+  }
+
+  async process(job: Job<ITxMonitorJob, any, string>): Promise<any> {
+    this.logger.log(`Monitoring job ${job.id} of type ${job.name}`);
+    try {
+      const numberBlocksToParse = 100n;
+      const txCapacityEpoch = job.data.epoch;
+      const previousKnownBlockNumber = (await this.blockchainService.getBlock(job.data.lastFinalizedBlockHash)).block.header.number.toBigInt();
+      const currentFinalizedBlockNumber = await this.blockchainService.getLatestFinalizedBlockNumber();
+      const blockList: bigint[] = [];
+      for (let i = previousKnownBlockNumber; i <= currentFinalizedBlockNumber && i < previousKnownBlockNumber + numberBlocksToParse; i += 1n) {
+        blockList.push(i);
+      }
+      const txBlockHash = await this.crawlBlockList(job.data.txHash, txCapacityEpoch, blockList);
+
+      if (txBlockHash) {
+        this.logger.verbose(`Successfully completed job ${job.id}`);
+        return { success: true };
+      }
+      return { success: false };
+    } catch (e) {
+      this.logger.error(`Job ${job.id} failed (attempts=${job.attemptsMade}) with error: ${e}`);
+      throw e;
+    } finally {
+      // do some stuff
+    }
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  @OnWorkerEvent('completed')
+  onCompleted() {
+    // do some stuff
+  }
+
+  private async crawlBlockList(txHash: Hash, epoch: string, blockList: bigint[]): Promise<BlockHash | undefined> {
+    // Use Promise.all to await all filter operations before returning undefined
+    await Promise.all(
+      blockList.map(async (blockNumber) => {
+        const blockHash = await this.blockchainService.getBlockHash(blockNumber);
+        const block = await this.blockchainService.getBlock(blockHash);
+        const txInfo = block.block.extrinsics.filter((extrinsic) => extrinsic.hash.toString() === txHash.toString());
+        this.logger.debug(`Extrinsics: ${block.block.extrinsics[0]}`);
+
+        if (txInfo.length > 0) {
+          this.logger.debug(`Found tx ${txHash} in block ${blockNumber}`);
+          const at = await this.blockchainService.api.at(blockHash.toHex());
+          const events = await at.query.system.events();
+          events.subscribe((records) => {
+            records.forEach(async (record) => {
+              const { event } = record;
+              const eventName = event.section;
+              const { method } = event;
+              const { data } = event;
+              this.logger.debug(`Received event: ${eventName} ${method} ${data}`);
+              if (eventName.search('capacity') !== -1 && method.search('Withdrawn') !== -1) {
+                const capacityWithDrawn = BigInt(data[1].toString());
+                this.logger.debug(`Capacity withdrawn: ${capacityWithDrawn}`);
+                this.setEpochCapacity(epoch, capacityWithDrawn);
+              }
+            });
+          });
+        }
+      }),
+    );
+
```

Comment:
this function parses possible 100 blocks in future from last known finalized block at time of send.
If it finds the tx hash in any block, goes to events get capacity and done